### PR TITLE
Don't include dev deps in Index#dependency_names

### PR DESCRIPTION
**Pull request by segiddins**
See #5256

### CODE REVIEW

**Comment by segiddins** [comment]
@indirect r?

**Comment by indirect** [comment]
LGTM. 👍 Why do we need this change? What issue does it resolve?

File: lib/bundler/index.rb [hunk 1]
```diff
@@ -109,14 +109,19 @@ def each(&blk)
 
     # returns a list of the dependencies
     def unmet_dependency_names
-      names = dependency_names
-      names.delete_if {|n| n == "bundler" }
-      names.select {|n| search(n).empty? }
+      dependency_names.select do |name|
+        name != "bundler" && search(n).empty?
```
Comment (Paxa) [inline lib/bundler/index.rb]: Tests fails because can't fine variable `n`

**Comment by segiddins** [comment]
@indirect it just potentially saves downloading gems

**Comment by segiddins** [comment]
@bundlerbot r+

**Comment by segiddins** [comment]
@bundlerbot retry

**Comment by Paxa** [comment]
Thank you

**Pull request merged**